packages/next/next-server/server/next-server.ts

```diff
@@ -1010,6 +1010,23 @@ export default class Server {
       return { html, pageData, sprRevalidate }
     })
 
+    // render fallback if cached data wasn't available
+    if (!isResSent(res) && !isDataReq && isDynamicRoute(pathname)) {
```

Comment:
In production this should be serving the static HTML file from build, not rendering the fallback on-demand.

Do we handle this behavior somewhere that I'm missing?

Reply:
Updated to load the pre-built fallback in production mode